### PR TITLE
[azopenaiassistants] Regenerating to work with the new assistants v2 (beta) endpoint

### PR DESCRIPTION
**Pull request by richardpark-msft**
Regenerating based on the Assistants V2 update (it's beta, just v2 of their beta).

The non-autogenerated code is there to work around the way that `union` translates to swagger - it typically produces the leaf objects but the union itself is empty. model_custom.go are manual representations of the unions, themselves.
    
I've also addressed some linter errors, removing log.Fatal's from the examples.

NOTE: this is just to get some of the simpler-to-review changes out of the way. I will make another PR after this to address writing examples for some of the new features, and to introduce streaming.

### CODE REVIEW

File: sdk/ai/azopenaiassistants/models_custom.go [hunk 1]
```diff
@@ -0,0 +1,197 @@
...
+	CodeInterpreterToolDefinition *CodeInterpreterToolDefinition
+
+	// FileSearchToolDefinition should be set if you attach files for the FileSearchToolDefinition tool.
+	FileSearchToolDefinition *FileSearchToolDefinition
+}
+
+// MarshalJSON implements the json.Marshaller interface for type MessageAttachmentToolDefinition.
+func (m *MessageAttachmentToolDefinition) MarshalJSON() ([]byte, error) {
+	switch {
+	case m.CodeInterpreterToolDefinition != nil:
```
Comment (jhendrixMSFT) [inline sdk/ai/azopenaiassistants/models_custom.go:27]: This won't handle the case when sending a JSON `null` (unclear if that's important). See the `populate()` func that gets generated in the `models_serde.go` file.

File: sdk/ai/azopenaiassistants/models_custom.go [hunk 2]
```diff
@@ -0,0 +1,197 @@
...
+		return json.Marshal(m.FileSearchToolDefinition)
+	default:
+		return nil, errors.New("neither CodeInterpreterToolDefinition or FileSearchToolDefinition were set")
+	}
+}
+
+// UnmarshalJSON implements the json.Unmarshaller interface for type MessageAttachmentToolDefinition.
+func (m *MessageAttachmentToolDefinition) UnmarshalJSON(data []byte) error {
+
+	var v *struct {
```
Comment (jhendrixMSFT) [inline sdk/ai/azopenaiassistants/models_custom.go:39]: Is this intermediate type really necessary? It looks to be identical to `MessageAttachmentToolDefinition`.

**Review by jhendrixMSFT — APPROVED** [review]

File: sdk/ai/azopenaiassistants/models_custom.go (hunk 2 above)
Comment (richardpark-msft) [reply sdk/ai/azopenaiassistants/models_custom.go:39]: I deserialized this incorrectly - they're not sub-objects, it's a union. I'll fix it.

File: sdk/ai/azopenaiassistants/models_custom.go (hunk 1 above)
Comment (richardpark-msft) [reply sdk/ai/azopenaiassistants/models_custom.go]: Fixed with a little if/else so we still call json.Marshal(). The service does validation on the value, so there's no reason for me to.

File: sdk/ai/azopenaiassistants/models_custom.go (hunk 2 above)
Comment (richardpark-msft) [reply sdk/ai/azopenaiassistants/models_custom.go:36]: Fixed.

**Review by weshaggard — APPROVED** [review]

**Pull request merged**